sentry-ruby/lib/sentry/std_lib_logger.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Sentry
+	# Ruby Logger support Add commentMore actions
+	# intercepts any logger instance and send the log to Sentry too.
+	module StdLibLogger
+
+		SEVERITY_MAP = {
+		  0 => :debug,
+		  1 => :info,
+		  2 => :warn,
+		  3 => :error,
+		  4 => :fatal
+		}.freeze
+
+		def add(severity, message = nil, progname = nil, &block)
+
+			return unless Sentry.initialized? && Sentry.get_current_hub
+
+			config = Sentry.configuration
+			return unless config.enable_logs && config.send_stdlib_logs
```

Comment:
This guard won't be needed when you add patching based on our `enabled_patches` option like I mentioned in the other comment. When registering your patch via `register_patch` you'll be able to do something like:

```ruby
Sentry.register_patch(:logger, Sentry::StdLibLogger, ::Logger) do |config|
  if config.enable_logs
    ::Logger.prepend(Sentry::StdLibLogger)
  else
    Sentry.sdk_logger.warn(":logger patch enabled but `enable_logs` is turned off - skipping applying patch")
  end
end
```